### PR TITLE
gh-107901: compiler replaces POP_BLOCK instruction by NOPs before optimisations

### PR DESCRIPTION
**Pull request by iritkatriel**
The `POP_BLOCK` are not needed once the exception handlers are attached to each instruction, so we can replace them by NOPs before optimisation. (The block push instruction are still needed for reachability analysis).

* Issue: gh-107901

### CODE REVIEW

**Comment by iritkatriel** [comment]
Converted to draft - I think we should merged https://github.com/python/cpython/pull/114535 and then this one becomes simpler.

File: Python/flowgraph.c [hunk 1]
```diff
@@ -953,6 +953,15 @@ label_exception_targets(basicblock *entryblock) {
            PyMem_Free(except_stack);
         }
     }
+
+    for (basicblock *b = entryblock; b != NULL; b = b->b_next) {
+        for (int i = 0; i < b->b_iused; i++) {
+            cfg_instr *instr = &b->b_instr[i];
+            if (instr->i_opcode == POP_BLOCK) {
+                INSTR_SET_OP0(instr, NOP);
```
Comment (markshannon) [inline Python/flowgraph.c]: Could we do this in the main loop, at line 905?

Comment (iritkatriel) [reply Python/flowgraph.c]: Probably. I'll try that.

File: Python/flowgraph.c [hunk 2]
```diff
@@ -953,6 +954,7 @@ label_exception_targets(basicblock *entryblock) {
            PyMem_Free(except_stack);
         }
     }
+
```
Comment (iritkatriel) [inline Python/flowgraph.c]:
```suggestion
```

File: Python/flowgraph.c (hunk 1 above)
Comment (iritkatriel) [reply Python/flowgraph.c]: Yup, it works.

**Pull request merged**